### PR TITLE
Fix the nnops to use the cuda version

### PR DESCRIPTION
**Pull request by xiki-tempula**
## Description
Fix the dependencies by force the installation to be on CUDA.
Fix a bug in the getting box_length.
Fix the test.

## Status
- [ ] Ready to go

### CODE REVIEW

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -41,6 +41,8 @@ jobs:
 
     # More info on options: https://github.com/conda-incubator/setup-miniconda
     - uses: mamba-org/provision-with-micromamba@main
+      env:
```
Comment (wiederm) [inline .github/workflows/CI.yaml:44]: that is great! is there documentation for this?

**Review by wiederm — APPROVED** [review]
All of this looks good! Thanks for the changes.
Tests are still not passing though. The error looks like a GH permission error, I will investigate. As soon as tests are running I will accept this PR.

File: endstate_correction/neq.py [hunk 2]
```diff
@@ -70,14 +71,14 @@ def perform_switching(
         # select a random frame
         random_frame_idx = random.randint(0, len(samples.xyz) - 1)
         # select the coordinates of the random frame
-        coord = samples.xyz[random_frame_idx]
+        coord = samples.openmm_positions(random_frame_idx)
         if samples.unitcell_lengths is not None:
-            box_length = samples.unitcell_lengths[random_frame_idx]
+            box_length = samples.openmm_boxes(random_frame_idx)
```
[the remark this answers is not in the document]
Comment (xiki-tempula) [reply endstate_correction/neq.py:76]: This is a bug fix, I think it won't work in the original form.

File: .github/workflows/CI.yaml (hunk 1 above)
Comment (wiederm) [reply .github/workflows/CI.yaml:44]: cheers!

**Pull request closed without merging**